core-java/src/main/java/eu/ostrzyciel/jelly/core/internal/ProtoDecoderImpl.java

```diff
@@ -0,0 +1,292 @@
+package eu.ostrzyciel.jelly.core.internal;
+
+import static eu.ostrzyciel.jelly.core.JellyOptions.*;
+
+import eu.ostrzyciel.jelly.core.*;
+import eu.ostrzyciel.jelly.core.proto.v1.Rdf;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+
+public sealed class ProtoDecoderImpl<TNode, TDatatype, TTriple, TQuad, TOut>
+    extends ProtoDecoder<TNode, TDatatype, TTriple, TQuad, TOut> {
+
+    protected final BiConsumer<String, TNode> namespaceHandler;
+    private Rdf.RdfStreamOptions supportedOptions;
+
+    public ProtoDecoderImpl(
+        Class<TDatatype> datatypeClass,
+        ProtoDecoderConverter<TNode, TDatatype, TTriple, TQuad> converter,
+        NameDecoder<TNode> nameDecoder,
+        BiConsumer<String, TNode> namespaceHandler,
+        Rdf.RdfStreamOptions supportedOptions
+    ) {
+        super(datatypeClass, converter, nameDecoder);
+        this.namespaceHandler = namespaceHandler;
+        this.supportedOptions = supportedOptions;
+    }
+
+    @Override
+    protected int getNameTableSize() {
+        return Optional.ofNullable(supportedOptions)
```

Comment:
Does Java 17 have a null coalescing operator? If not, I'd prefer an if == null here, really...

Reply:
des decision: do not make use of Optional, due to pef

Reply:
doned